lib/sparkle_formation/utils.rb

```diff
@@ -3,6 +3,43 @@ class SparkleFormation
   # Helper utilities
   module Utils
 
+    module TypeCheckers
+
+      # Validate given value is type defined within valid types
+      #
+      # @param val [Object] value
+      # @param types [Class, Array<Class>] valid types
+      # @return [NilClass]
+      # @raises [TypeError]
+      def __t_check(val, types)
+        types = [types] unless types.is_a?(Array)
+        if(types.none?{|t| val.is_a?(t)})
+          file_name, line_no = ::Kernel.caller.detect do |l|
+            !l.split(':').first.to_s.include?('/sparkle_formation')
```

Comment:
it would still behave exactly the same as the current code / have the same false positive

```
l = '/foo/bar/sparkle_formation/xyz.rb:123:in `execute'
l.include?('/sparkle_formation')
l.split(':').first.to_s.include?('/sparkle_formation')
```